Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -217,6 +218,12 @@ class DatabaseContainer: NSPersistentContainer {
     func write(_ actions: @escaping (DatabaseSession) throws -> Void, completion: @escaping (Error?) -> Void) {
         writableContext.perform {
             log.debug("Starting a database session.", subsystems: .database)
+            guard self.canWriteData else {
+                log.debug("Discarding write attempt.", subsystems: .database)
+                completion(nil)
+                return
+            }
```

Comment:
Added a safeguard here for cases where batch delete is in process and some delayed action triggers database write.